connaisseur/keys.py

```diff
@@ -0,0 +1,141 @@
+import base64
+import hashlib
+import re
+import binascii
+
+import ecdsa
+import rsa
+from pyasn1.error import PyAsn1Error
+
+from connaisseur.image import Image
+from connaisseur.exceptions import NoSuchClassError, WrongKeyError
+
+KMS_REGEX = r"(?:awskms|gcpkms|azurekms|hashivault|k8s):\/{2,3}[a-zA-Z0-9_.+\/:-]+"
+KEYLESS_REGEX = r"^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$"
+
+
+class Key:
+    """
+    Abstract Key class, containing the public key used for verifying the image signatures.
+    """
+
+    value: str
+
+    def __new__(cls, data: str):
+        try:
+            key_cls = Key.__get_key_type_cls(data)
+            return super(Key, cls).__new__(key_cls)
+        except Exception as err:
+            msg = "Unable to find find class {class_name}."
+            raise NoSuchClassError(message=msg, class_name=key_cls) from err
+
+    @staticmethod
+    def __get_key_type_cls(data: str):
+        # key gets automatically identified
```

Comment:
If this is supposed to be a docstring, use triple quotes and command sentence, otherwise I'd say it's a superfluous comment